GOVCMS-6775: Add more functions to the banned list.

yusufhm opened the pull request:
Since the goal of the banned functions list is not to audit the code quality nor do syntax checks for the projects, but instead to only ensure no calls to banned functions are being made, I've disabled phpstan default checks, ~~as well as the ones from~~ removed the `ekino/phpstan-banned-code` extension, and used `spaze/phpstan-disallowed-calls` instead.

- ~~Removed phpstan extension installer so we can add ekino/phpstan-banned-code manually, while disabling all defaults.~~ Replaced `ekino/phpstan-banned-code` with `spaze/phpstan-disallowed-calls`, which allows for wildcard function names and method calls, which was missing in the former.
- Disabled default checks in phpstan.
- Added `mysql_*`, `pcntl_*` and a few other functions to the banned list.
- Added `mysqli::*()` & `SQLite3::*()` as banned method calls.

yusufhm committed:
GOVCMS-6775: Add more functions to the banned list.

- Removed phpstan extension installer so we can add ekino/phpstan-banned-code manually, while disabling all defaults.
- Disabled default checks in phpstan.
- Manually added ekino/phpstan-banned-code.
- Added mysql_*, pcntl_* and a few others to the banned list.

steveworley commented on phpstan.neon:
```diff
@@ -1,38 +1,49 @@
     - inc
   reportUnmatchedIgnoredErrors: false
   ignoreErrors:
-    - "#Function (.*) not found#"
```
This looks fine to me, but we might want to trial it on a few canaries just to see how this impacts some real use cases.

stooit reviewed (approved):
Ditto, suggest we merge this and have the canaries run through. There are several that fail at the moment due to false positives - would expect a lot more green results tomorrow.

yusufhm committed:
GOVCMS-6775: Use spaze/phpstan-disallowed-calls.

- Allows specifying wildcards for functions.
- Also has disallowedMethodCalls, which covers our needs.
